opend/src/opend.d

```diff
@@ -0,0 +1,160 @@
+module opend;
+
+import std.algorithm;
+import std.getopt;
+import std.file;
+import std.string : endsWith;
+import std.path;
+import std.process : execute;
+import std.stdio : writeln;
+
+string compiler = `dmd`;
+string type = "executable";
+string buildType = "debug";
+
+/// Retrieves all files as full path
+string[] getFiles(string dir)
+{
+    string[] ret;
+    foreach(ref DirEntry d; dirEntries(dir, SpanMode.breadth))
+    {
+        // Only care about files with ".d" extensions
+        if(d.isDir || extension(d.name) != ".d")
+            continue;
+
+        // Auto-escape pathes!
+        // Actually, I checked (on Windows) and paths having spaces in them work fine too.
+        ret ~= /*"\"" ~ */absolutePath(d.name) /*~ "\""*/;
+    }
+    return ret;
+}
+
+string getExeFileName()
+{
+    immutable dname = baseName(getcwd());
+    writeln(dname);
+    version(Windows)
+    {
+        immutable extension = ".exe";
+    }
+    else
+    {
+        immutable extension = "";
+    }
+
+    return dname ~ extension;
+}
+
+string getLibFileName()
+{
+    immutable dname = baseName(getcwd());
+    version(Windows)
+    {
+        immutable extension = ".lib";
+    }
+    else
+    {
+        immutable extension = "";
+    }
+
+    return dname ~ extension;
+}
+
+string[] getDefaultFiles()
+{
+    if(exists("src") && isDir("src"))
+    {
+        return getFiles("src");
+    }
+    else if(exists("source") && isDir("source"))
+    {
+        return getFiles("source");
+    }
+    return [];
+}
+
+string[] extraSwitches = ["-od=build/"];
+
+string flattenCmd(string[] cmd)
+{
+    string ret;
+    foreach(str; cmd)
+    {
+        ret ~= str ~ " ";
+    }
+    return ret;
+}
+
+string getOfFlag()
+{
+    // With -lib enabled, dmd puts .lib file wherever the -od is specified. Doesn't happen with executables
+    if(type == "library")
+        return "-of=" ~ getLibFileName();
+    else
+        return "-of=build/" ~ getExeFileName();
+}
+
+int main(string[] args)
+{
+    auto optionsResult = getopt(args,
+        "compiler", &compiler,
+        "output", &type,
+        "type", &buildType);
+
+    static immutable availOutputs = ["executable", "library"];
+    static immutable availBuildTypes = ["debug", "release"];
+
+    if(availOutputs.all!(x => x != type))
+    {
+        writeln(i"Unsupported output type `$(type)`. Only `executable` and `library` are supported");
```

Comment:
"Output" rather than "output type" makes more sense here
```suggestion
        writeln(i"Invalid output `$(type)` (provide `executable` or `library`)");
```